flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/AWSUtil.java

```diff
@@ -50,8 +50,14 @@ public static AmazonKinesisClient createKinesisClient(Properties configProps) {
 		awsClientConfig.setUserAgent("Apache Flink " + EnvironmentInformation.getVersion() +
 			" (" + EnvironmentInformation.getRevisionInformation().commitId + ") Kinesis Connector");
 
-		AmazonKinesisClient client =
-			new AmazonKinesisClient(AWSUtil.getCredentialsProvider(configProps).getCredentials(), awsClientConfig);
+		AmazonKinesisClient client;
+		if (AWSUtil.getCredentialsProvider(configProps) != null) {
+			client = new AmazonKinesisClient(
+				AWSUtil.getCredentialsProvider(configProps).getCredentials(), awsClientConfig);
```

Comment:
Ah, sorry, I made a misjudgement on my last comment here. It'll actually be better to have `AWSUtil.getCredentialsProvider(configProps)` declared as a variable outside and reuse it in the `if` clause, otherwise we'll be calculating the credentials provider twice. Trivial thing, since it's only a one-time thing, but it'll be good to revert it back to what you did in the beginning :)